### PR TITLE
fix(vouchers): use the documented hotspot voucher routes and shape

### PR DESCRIPTION
**Pull request by jpennin5**
## Description

Part of #108 (items B1 and A3).

**Problem:** every voucher tool calls
`/integration/v1/sites/{site}/vouchers`, which no controller serves, so
none of them has ever succeeded.

**Root cause:** the documented Integration v1 surface is
`/v1/sites/{siteId}/hotspot/vouchers`, and the `Voucher` model described a
legacy shape the endpoint never returns (`_id`, `site_id`, `status`,
`duration`, `create_time`, string `code`).

**Fix (two commits):**

1. Routes and model moved to the documented surface. The model mirrors
   the documented response: camelCase aliases, only `id` required,
   everything else optional and omitted when absent (the convention
   settled in #103) — `activatedAt` in particular does not exist until a
   voucher is first used. `create_vouchers` sends the documented request
   body (`name`, `timeLimitMinutes`; the previous body sent `duration` in
   seconds and `qos_*` limits the endpoint never accepted).
   `bulk_delete_vouchers` reads the documented `{"vouchersDeleted": N}`
   response and refuses an empty filter rather than passing a
   delete-everything ambiguity to the controller.
2. Live verification found two divergences from the docs, both handled:
   the 201 generation body nests the batch under a `"vouchers"` key
   (reads return the objects directly, so both shapes are unwrapped), and
   `code` arrives as a string where the docs say integer, confirming the
   `int | str` typing.

**Live verification (Network 10.5.67):** full round-trip — vouchers
created, listed, fetched, and deleted against the documented routes.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

The `create_vouchers` signature changes, but the previous signature could
never produce a successful call, so nothing that worked can break.

## Testing

- Test fixtures rebuilt to the documented camelCase shape, including the
  real 201 `{"vouchers": [...]}` nesting and string `code` regression
  cases; empty-filter bulk delete rejection covered.
- `pytest tests/unit/` and `pre-commit run --all-files` pass.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (CHANGELOG.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published (none — stands alone)

## AI Assistance

This PR was created with assistance from Claude Code.

**Human Review Status:** ✅ Reviewed and approved by @jpennin5
**Testing:** Full unit suite passing; create/list/get/delete round-trip
verified against a live Network 10.5.67 controller.
**Security Review:** Completed — all fixture identifiers are synthetic;
live-created test vouchers were deleted.

### CODE REVIEW

File: src/tools/vouchers.py [hunk 1]
```diff
@@ -43,18 +75,21 @@ async def list_vouchers(
...
             params["filter"] = filter_expr
 
-        response = await client.get(f"/integration/v1/sites/{site_id}/vouchers", params=params)
-        data = response if isinstance(response, list) else response.get("data", [])
+        response = await client.get(
+            f"/integration/v1/sites/{site_id}/hotspot/vouchers", params=params
+        )
 
-        return [Voucher(**voucher).model_dump() for voucher in data]
+        return [
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/vouchers.py:82]: Fixed in 14c5f17: all five endpoints are built via `settings.get_integration_path()`, with a cloud-v1 path-shape regression test asserting `/v1/sites/{site}/hotspot/vouchers`.

File: src/tools/vouchers.py [hunk 2]
```diff
@@ -66,80 +101,93 @@ async def get_voucher(site_id: str, voucher_id: str, settings: Settings) -> dict
...
+        Created vouchers, including their access codes
     """
     validate_confirmation(confirm, "create vouchers", dry_run)
 
+    if not name:
+        raise ValidationError("Voucher name is required")
+    if time_limit_minutes < 1:
+        raise ValidationError("time_limit_minutes must be at least 1")
+    if not 1 <= count <= 1000:
+        raise ValidationError("count must be between 1 and 1000")
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/vouchers.py:156]: Added in 14c5f17 — the documented bounds (time_limit_minutes 1–1,000,000; guest limit ≥ 1; data 1–1,048,576 MB; rates 2–100,000 kbps) now fail fast with ValidationError.

File: tests/unit/tools/test_vouchers_tools.py [hunk 3]
```diff
@@ -33,395 +40,305 @@ def mock_settings():
...
-    mock_client.get = AsyncMock(return_value=mock_response)
-    mock_client.__aenter__ = AsyncMock(return_value=mock_client)
-    mock_client.__aexit__ = AsyncMock(return_value=None)
+    """The documented payload parses and the hotspot path is called."""
+    second = {**sample_voucher, "id": "0b1c2d3e-1111-2222-3333-444455556666", "code": 111222333}
+    client = _make_client({"data": [sample_voucher, second]})
 
-    with patch.object(vouchers_module, "UniFiClient", return_value=mock_client):
+    with patch.object(vouchers_module, "UniFiClient", return_value=client):
         result = await list_vouchers("default", mock_settings)
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply tests/unit/tools/test_vouchers_tools.py:85]: Restored in 14c5f17 with `test_list_vouchers_authenticates_when_needed`.

**Comment by enuno** [comment]
### Summary
Request changes. The PR is not yet merge-ready because the current status checks are not clean.

### Critical findings
- Merge state: \ / \.
- Failing check(s): Mermaid Diagram Sync Assistant.
- The changed tooling surface should be revalidated after the failing check is resolved.

### Suggestions
1. Resolve the failing status check(s) and rerun CI.
2. Reconfirm the changed tooling behavior against the intended controller/fixture shapes.
3. Ensure regression tests cover the original failure mode and edge cases introduced here.
4. Update docs if any request/response or error semantics changed.

### Verification checklist for the author
- [ ] Required checks are passing
- [ ] Regression tests cover the changed path
- [ ] Lint / format / type checks pass locally
- [ ] Documentation is updated where user-facing behavior changed
- [ ] No secrets, credentials, or controller-identifying data were introduced

**Comment by jpennin5** [comment]
The failing check is the `Mermaid Diagram Sync Assistant` GitHub App, not a workflow in this repository. Its check-run output on this PR reads: "Webhook Processing Failed — An unexpected error occurred while processing the webhook. Please try pushing a new commit or check your repository configuration." That processing happens on the app's side; pushing new commits to these PRs has re-triggered the check with the same result, so it is not reachable from a fork.

This is the loop documented in #112: the app has errored on every fork PR since late June, fork CI runs sit at `action_required` awaiting maintainer approval, and the daily automated review then reads the app error as a failing required check and requests changes.

Every check that can conclude on a fork PR is green here (both Socket Security checks). The remaining steps are repo-side — approving the CI runs and either repairing/removing the app integration or unrequiring its check. #117 merged with the identical check state, so the merge path exists. Happy to rebase or adjust anything code-side the moment CI can actually run.
